doc/api/stream.md

```diff
@@ -154,89 +325,134 @@ Examples of readable streams include:
 * [child process stdout and stderr][]
 * [`process.stdin`][]
 
-#### Event: 'close'
+All [Readable][] streams implement the interface defined by the
+`stream.Readable` class.
+
+#### Two Modes
+
+Readable streams may operate in one of two modes: flowing and paused.
+
+When in flowing mode, data is read from the underlying system automatically
+and provided to an application as quickly as possible using events via the
+[`EventEmitter`][] interface.
+
+In paused mode, the [`stream.read()`][stream-read] method must be called
+explicitly to read chunks of data from the stream.
+
+All [Readable][] streams begin in paused mode but can be switched to flowing
+mode in one of the following ways:
+
+* Adding a [`'data'`][] event handler.
+* Calling the [`stream.resume()`][stream-resume] method.
+* Calling the [`stream.pipe()`][] method to send the data to a [Writable][].
+
+The Readable can switch back to paused mode using one of the following:
+
+* If there are no pipe destinations, by calling the
+  [`stream.pause()`][stream-pause] method.
+* If there are pipe destinations, by removing any [`'data'`][] event
+  handlers, and removing all pipe destinations by calling the
+  [`stream.unpipe()`][] method.
+
+*Note*: For backwards compatibility reasons, removing [`'data'`][] event
+handlers will **not** automatically pause the stream. Also, if there are piped
+destinations, then calling [`stream.pause()`][stream-pause] will not guarantee
+that the stream will *remain* paused once those destinations drain and ask for
+more data.
 
-Emitted when the stream and any of its underlying resources (a file
-descriptor, for example) have been closed. The event indicates that
-no more events will be emitted, and no further computation will occur.
+*Note*: If a [Readable][] is switched into flowing mode and there are no event
+handlers or [`stream.pipe()`] destinations attached to handle the data, that
+data will be lost.
```

Comment:
I do not think this is accurate for streams3. In streams3, there is little distinction between the two modes, which in fact are two states. The streams can be either flowing or paused. Flowing means that data is being processed, paused that no data is being processed.
The flowing state is part of `pipe`, see https://github.com/nodejs/node/blob/master/lib/_stream_readable.js#L605-L609 and https://github.com/nodejs/node/blob/master/lib/_stream_readable.js#L552-L569.

This means that if there is a `'data'` handler and a piped destination, the `'data'` events will be emitted at the speed of the slowest of the piped destinations.